extensions/applicationinsights-debugplugin-js/README.md

```diff
@@ -0,0 +1,196 @@
+have a `[-]` before it. Nodes like this also have their type, that being object or array, denoted
+next to them with `{}` or `[]` respectively. Between these braces/brackets will be a number.
+This number shows the number of entries within that object or array.
+
+There a button above these entries that says "copy current node." This button will, if a node is
+selected, copy the node and its children up to a certain depth to the clipboard as a JSON string.
+This will also handle circular references.
+
+## When Should This Plugin Be Used
+
+This plugin is used when:
+- data isn't reaching azure
+- things don't appear to be tracked
+- something is crashing and developers are unsure if the plugin is causing it
+
+## Where to get More Information
+
+TBD
+
+## Why This Plugin Exists
+
+ApplicationInsights-js has lacked a solid way to debug issues for awhile now. Often, developers
+had to dig through code and go to a multitude of different areas to even begin diagnosing an issue.
+
+The impact of all of these will bring about a much better development experience for those using
+ApplicationInsights-js. Just the fact that they are able to see when a function is fired will
+dramatically cut-down on time spent diagnosing and eliminating bugs. The tree view will foster
+communication between developers and testers, allowing them to copy and send important pieces of data
+elsewhere.
+
+Instead of needing to come to the ApplicationInsights-js team with every bug encountered, in many
+cases developers will be able to use these tools to diagnose things that are going wrong on their own
+thanks to this new layer of transparency.
+
+## Getting Started
+
+run the following console command to install appInsights and the debug plugin:
+
+```zsh
+npm install --save @microsoft/applicationinsights-debugplugin-native @microsoft/applicationinsights-web
+```
+
+## Initializing the Plugin
+
+```js
+import { ApplicationInsights } from '@microsoft/applicationinsights-web';
+import { DebugPlugin } from '@microsoft/applicationinsights-debugplugin-js';
+
+const toTrack = [
+    'trackEvent',
+    'trackPageView',
+    'trackPageViewPerformance',
+    'trackException',
+    'trackTrace',
+    'trackMetric',
+    'trackDependencyData',
+    'throwInternal',        // called when a message is logged internally
+    'logInternalMessage',   // called when a message is logged internally
+    'triggerSend',          // called when data is queued to be sent to the server
+    '_sender',              // called when data is sent to the server
+];
+
+const debugPluginInstance = new DebugPlugin();
+const appInsights = new ApplicationInsights({
+    config: {
+        instrumentationKey: 'YOUR_INSTRUMENTATION_KEY_GOES_HERE',
+        extensions: [debugPluginInstance],
+        extensionConfig: {
+            [DebugPlugin.identifier]: {
+                trackers: toTrack
+            }
+        }
+    }
+});
+appInsights.loadAppInsights();
+```
+
```

Comment:
Next PR we should include details about the config, in a table like some of the other readme's